lang/gcc14/Portfile

```diff
@@ -394,6 +417,14 @@ if {${subport} eq ${libgccname}} {
     # Activate hack for new libgcc
     # https://trac.macports.org/wiki/PortfileRecipes#deactivatehack
     pre-activate {
+        if {![catch {set installed [lindex [registry_active libgcc7] 0]}]} {
+            # Extract the epoch of the installed libgcc7
+            set _epoch [lindex $installed 5]
+            # If < 4 need to deactivate
```

Comment:
@kencu Does this mean that we actually need to bump the epoch of libgcc7?

I think this hack is why I had it bumped previously.

Reply:
no, we're OK with that. libgcc7 is force-deactivated when libgcc14 is called in as the base, and the revbump to libgcc7 works properly now and installs the reduced libgcc7 port.

Reply:
Oh, now hold on... I might be wrong here and might have spoken too soon.

Let me look at the last time I did this. You may be right, and we may need to bump the epoch.

Reply:
You don't need to bump the epoch. You only bump the epoch when the versioning scheme changes such that the new version looks older than the old one.

You shouldn't be looking at the epoch for the deactivate hack. Look at the version and/or revision.

Reply:
All the other deactivate hacks in this file are looking at epoch. Shall I change just the gcc7 one?

Reply:
Oh, I remember why we used epich. The libgcc version was not changing, so we couldn’t use version. I think there was no way to test for a specific revision, so we used epoch instead.

Now we’ll have to see if we can test for revision, I guess. 

If not, the we have to use epoch, despite what Josh says, unless someone has a brilliant idea.

Reply:
I don't know why you wouldn't be able to check the revision. The value returned by `registry_active` lists name,version,revision,variants,active,epoch.

Reply:
OK, to do this right, we’ll need to generate a tcl function “newer” that looks at epoch, version, and revision, and returns a boolean if those are “newer” than a specified epoch, version, and revision.

Reply:
Well, what if someone bumps the epoch on some gcc version someday for some reason though?

We would expect to use the same test for all libgcc versions, forever, right?

Reply:
```
$epoch < E || [vercmp $_vers < X.Y] || $_vers eq "X.Y" && $_rev < Z
```

Reply:
If that makes you feel better, sure that will work. But it seems extremely unlikely that gcc will ever change its versioning scheme in a minor release, which is the only reason the epoch would ever be relevant.

Reply:
I've pushed an updated implementation.

I've found the definition of a registry entry here:
https://github.com/macports/macports-base/blob/1b43b563cf14d7b1b2e3d6d43e1c76de577f218a/src/registry2.0/registry.tcl#L58-L65 

This told me that version is field 1 and revision is field 2.

I've cargo-culted `[vercmp $_revision < 2]` rather than a simple `$_revision < 2`  because I wasn't sure whether the latter is correct / works in TCL.

PTAL

Reply:
`version` needs to be compared with `vercmp`, but `revision` and `epoch` are required to be simple integers.